### ISSUE TITLE
unit test flake: k8s.io/kubernetes/pkg/storage/etcd3

### ISSUE THREAD

**Issue by janetkuo** [kubernetes/kubernetes#24679]
~~https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/24138/kubernetes-pull-test-unit-integration/21794~~
Updated external link: https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/24138/kubernetes-pull-test-unit-integration/21794

```
FAIL	k8s.io/kubernetes/pkg/storage/etcd3	2.999s
```

```
=== RUN   TestWatch
--- FAIL: TestWatch (0.34s)
	watcher_test.go:204: #1: ResultChan should have been closed
```

Hit this in #24138

@kubernetes/sig-api-machinery

**Comment by hongchaodeng** [comment]
Logs?

**Comment by xiang90** [comment]
@janetkuo We probably should not paste google internal link here. People outside google cannot access it.

**Comment by lavalamp** [comment]
She did post the external link, is it not working for you?

**Comment by lavalamp** [comment]
The failure is "	watcher_test.go:204: #1: ResultChan should have been closed"

**Comment by xiang90** [comment]
@lavalamp No.

```
Use your SSO username and password
```

**Comment by xiang90** [comment]
@lavalamp It redirects me to `https://login.corp.google.com/xxx/uberproxy...`.

**Comment by lavalamp** [comment]
Oh, I see. @janetkuo when you visit the public link while logged into google corp, it redirects you to a non-google link, so copy-pasting that doesn't work.

Correct link should be: https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/24138/kubernetes-pull-test-unit-integration/21794

**Comment by xiang90** [comment]
Now I can see it. It is dup of https://github.com/kubernetes/kubernetes/issues/24528 I think. It is fixed yesterday. All prs that after the fix should be fine, existing pr will need to rebase to get rid of that flaky test.

**Comment by hongchaodeng** [comment]
The failure happened even before #24570, which should give more info for why Watch failed.

**Comment by fejta** [comment]
Assigning to @brendandburns per https://github.com/kubernetes/kubernetes/blob/master/pkg/OWNERS

Please address https://github.com/kubernetes/kubernetes/issues/26118 if you do not like this assignment.

**Comment by hongchaodeng** [comment]
This is already fixed by #24638.
Can we close this?

**Comment by wojtek-t** [comment]
Yeah - it was duplicate of #24528

Fixed by #24638
